### PR TITLE
libcaption: Fix header missing in install

### LINKED ISSUE #3925: /usr/include/obs/obs-internal.h references caption/caption.h but make install does not install that header
obs-studio 26.1.0

as the subject line says the headers are missing in the make install target or should not be pulled by the obs-internal header.

We found this by trying to build the obs gphoto2 plugin for openSUSE.

### PR DESCRIPTION
**Pull request by kkartaltepe**
### Description

libobs headers depend on libcaption headers. So ensure they are
installed with obs' headers.

### Motivation and Context

fixes #3925

### How Has This Been Tested?

Installing it lands in <caption/caption.h> as expected.

### Types of changes

 - Bug fix (non-breaking change which fixes an issue)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: deps/libcaption/CMakeLists.txt [hunk 1]
```diff
@@ -36,6 +36,7 @@ set(CAPTION_HEADERS
 )
 
 add_library(caption STATIC ${CAPTION_SOURCES})
+install_obs_headers("caption/caption.h")
```
Comment (darix) [inline deps/libcaption/CMakeLists.txt:39]: does this take care of headers pulled by caption.h ?

For me it pulled in more headers afterwards.

Comment (kkartaltepe) [reply deps/libcaption/CMakeLists.txt:39]: Doh, I did assume this was the only public header, I dont have a setup to compile an out of project plugin atm. Will update with all the headers. Thanks.

Comment (kkartaltepe) [reply deps/libcaption/CMakeLists.txt:39]: Looks like obs-internal was unintentionally leaked anyway. That should prevent the need for any of these deps headers.

Comment (darix) [reply deps/libcaption/CMakeLists.txt:39]: well if obs-internal will no longer be needed and with that the caption headers are no longer needed. I am fine too.

**Pull request merged**